gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -37,6 +40,40 @@ public static BrokerUserTaskCompletionRequest toUserTaskCompletionRequest(
             getDocumentOrEmpty(completionRequest, UserTaskCompletionRequest::getVariables),
             getStringOrEmpty(completionRequest, UserTaskCompletionRequest::getAction));
 
+    final String authorizationToken = getAuthorizationToken(context);
+    brokerRequest.setAuthorization(authorizationToken);
+
+    return brokerRequest;
+  }
+
+  public static BrokerUserTaskAssignmentRequest toUserTaskAssignmentRequest(
+      final UserTaskAssignmentRequest assignmentRequest,
+      final long userTaskKey,
+      final ServerWebExchange context) {
+
+    String action = getStringOrEmpty(assignmentRequest, UserTaskAssignmentRequest::getAction);
+    if (action.isBlank()) {
+      action = "assign";
+    }
+
+    final BrokerUserTaskAssignmentRequest brokerRequest;
```

Comment:
🔧 The only difference between the requests is the intent. Thus, you could determine the intent based on the condition and then create the request with the intent. Should be a bit less code and easier to see the difference between the requests.

Reply:
Refactoring done, if any suggestions, please let me know 🙂